File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Create a hashmap 
        HashMap<Integer, Integer> checkNum = new HashMap<>();

        // Go through each element of the array
        // Subtract the current element from the target to get new target
        // If the map contains this new target return the postion of each
        // Otherwise add that number and target to hashmap
        for (int i = 0; i < nums.length; i++) {
            int newTarget = target - nums[i];

            if (checkNum.containsKey(newTarget)) {
                return new int[]{checkNum.get(newTarget), i};
            }

            checkNum.put(nums[i], i);
        }

    return nums;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}

        for i in range(len(nums)):
            target_pair = target - nums[i]
            if target_pair in map:
                return [map.get(target_pair), i]
            else:
                map[nums[i]] = i

        